instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Singletons.java

```diff
@@ -42,7 +42,7 @@ public final class Servlet3Singletons {
       ResponseInstrumenterFactory.createInstrumenter(INSTRUMENTATION_NAME);
 
   private static final OutputStreamSnippetInjectionHelper SNIPPET_INJECTION_HELPER =
-      new OutputStreamSnippetInjectionHelper(() -> ExperimentalSnippetHolder.getSnippet());
+      new OutputStreamSnippetInjectionHelper(ExperimentalSnippetHolder.getSnippet());
```

Comment:
Overall this is a reasonable thing to do, but it causes test failures, see  https://scans.gradle.com/s/hosdpczhhhqck/tests/overview?outcome=flaky The problem is that in your test code you change the snippet, you need to rework the tests so that the snippet would be set only once during the startup as with this change calling `ExperimentalSnippetHolder.setSnippet` from test has no effect because `OutputStreamSnippetInjectionHelper` is already initialized with the old value.

Reply:
Thanks for the remind. I am working on it so I mark this as a draft pr. The plan would be give special jvm args for that particular test.